lib/wallaby/phantom/driver.ex

```diff
@@ -333,6 +333,135 @@ defmodule Wallaby.Phantom.Driver do
     end
   end
 
+  @doc """
+  Accept all JavaScript dialogs
+  """
+  def accept_dialogs(session) do
+    script = """
+    var page = this;
+    page.onAlert = function(msg) {}
+    page.onConfirm = function(msg) { return true; }
+    page.onPrompt = function(msg, defaultVal) { return defaultVal; }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+  end
+
+  @doc """
+  Dismiss all JavaScript dialogs
+  """
+  def dismiss_dialogs(session) do
+    script = """
+    var page = this;
+    page.onAlert = function(msg) {}
+    page.onConfirm = function(msg) { return false; }
+    page.onPrompt = function(msg, defaultVal) { return null; }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+  end
+
+  @doc """
+  Accept one alert triggered within `fun` and return the alert message.
+  """
+  def accept_alert(%Session{}=session, fun) do
+    script = """
+    var page = this;
+    page.__onAlertDefault = page.onAlert;
+    page.onAlert = function(msg) {
+      page.__alertMessage = msg;
+      page.onAlert = page.__onAlertDefault;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+    fun.(session)
+    script = """
+    var page = this;
+    page.onAlert = page.__onAlertDefault;
+    return ["ok", page.__alertMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
+  end
+
+  @doc """
+  Accept one confirm triggered within `fun` and return the confirm message.
+  """
+  def accept_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, true)
+  end
+
+  @doc """
+  Dismiss one confirm triggered within `fun` and return the confirm message.
+  """
+  def dismiss_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, false)
+  end
+
+  defp handle_confirm(%Session{}=session, fun, return_value) do
+    script = """
+    var page = this, returnVal = arguments[0];
+    page.__onConfirmDefault = page.onConfirm;
+    page.onConfirm = function(msg) {
+      page.__confirmMessage = msg;
+      page.onConfirm = page.__onConfirmDefault;
+      return returnVal;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script, [return_value])
+    fun.(session)
+    script = """
+    var page = this;
+    page.onConfirm = page.__onConfirmDefault;
+    return ["ok", page.__confirmMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
```

Comment:
@keathley what do you think about the caveat outlined in my code examples? Could be tricky to debug if you write it like in the second example.

We don't have any place to store session state except in the `session` struct, right? This change makes it the User's responsibility to reassign `session` to the updated struct that contains the messages.

If that's OK with you, I could start working on this. Should happen before the next release though, otherwise we'd immediately have incompatible changes in the release after that one. I'd be OK with directly putting it in this PR here.